File: PooPosting/PooPosting.View/src/app/Models/Dtos/LikeResult.ts
import {LikeDto} from "./LikeDto";
import {LikeState} from "../../Enums/LikeState";

export interface LikeResult {
  likeCount: number,
  dislikeCount: number,
  likes: LikeDto[],
  likeState: LikeState
}

File: PooPosting.Service/Models/LikeResult.cs
using PooPosting.Data.Enums;
using PooPosting.Service.Models.Dtos.Like;

namespace PooPosting.Service.Models;

public class LikeResult
{
    public LikeState LikeState { get; set; }
    public IEnumerable<LikeDto>? Likes { get; set; }
    public int LikeCount { get; set; }
    public int DislikeCount { get; set; }
}